pkg/sql/catalog/descs/collection.go

```diff
@@ -282,6 +284,29 @@ func (tc *Collection) GetAllDatabaseDescriptors(
 	return tc.kv.getAllDatabaseDescriptors(ctx, txn)
 }
 
+func (tc *Collection) GetAllDescriptorsInDatabase(
+	ctx context.Context, txn *kv.Txn, dbID descpb.ID,
+) ([]catalog.Descriptor, error) {
+	descs, err := tc.GetAllDescriptors(ctx, txn)
```

Comment:
Hydrating all descriptors just to get the subset of them that belong to this database is unfortunate. Leave a TODO to do something more efficient at some point.